contracts/SwapAndBridge.sol

```diff
@@ -0,0 +1,171 @@
+        uint256 minExpectedInputTokenAmount,
+        DepositData calldata depositData
+    ) external nonReentrant {
+        // @dev: Don't let caller use WETH, or better, hardcode the swap and acrossInput tokens to be USDC.e/USDC
+        // because 1Inch has special rules to handle WETH<-->ETH.
+
+        // Pull tokens from caller into this contract.
+        IERC20(swapToken).transferFrom(msg.sender, address(this), swapTokenAmount);
+
+        // Craft 1Inch swap arguments to swap `swapToken` for `acrossInputToken` using this contract as the recipient
+        // of the swap.
+        I1InchAggregationRouterV5.SwapDescription memory swapDescription = I1InchAggregationRouterV5.SwapDescription({
+            srcToken: IERC20(swapToken),
+            dstToken: IERC20(depositData.inputToken),
+            srcReceiver: payable(address(this)),
+            dstReceiver: payable(address(this)),
+            amount: swapTokenAmount,
+            minReturnAmount: minExpectedInputTokenAmount,
+            flags: 0 // TODO: Change this if we want to support ETH swaps and/or partial fills (i.e.
+            // swapTokenAmount != spentAmount). For now, set to 0 for simplicity.
+        });
+
+        // Swap and run safety checks.
+        uint256 srcBalanceBefore = IERC20(swapToken).balanceOf(address(this));
+        uint256 dstBalanceBefore = IERC20(depositData.inputToken).balanceOf(address(this));
+
+        // @dev: Example swap I used for comparison:
+        // https://optimistic.etherscan.io/tx/0x8a4e77ee1a62e94b42b21e849bcdabd60d43ac49191cd2878f6b47f395f26abc
+        IERC20(depositData.inputToken).safeIncreaseAllowance(address(oneInchRouter), swapTokenAmount);
+        (uint256 returnAmount, ) = oneInchRouter.swap(
+            aggregationExecutor,
+            swapDescription,
+            new bytes(0), // TODO: No IERC20Permit.permit needed since we're sending an approval?
+            new bytes(0) // TODO: We don't want to execute any data on swaps but I'm not sure how this is used.
+        );
+
+        // Sanity check that we received exactly as much as the oneInchRouter said we did.
+        require(
+            returnAmount == dstBalanceBefore - IERC20(depositData.inputToken).balanceOf(address(this)),
+            "return amount"
+        );
+        // Sanity check that received amount from swap is enough to submit Across deposit with.
+        require(returnAmount >= minExpectedInputTokenAmount, "min expected input amount");
+        // Sanity check that we don't have any leftover swap tokens that would be locked in this contract (i.e. check
+        // that we weren't partial filled).
+        require(
+            srcBalanceBefore - IERC20(swapToken).balanceOf(address(this)) == swapTokenAmount,
+            "leftover src tokens"
+        );
+
+        // Deposit the swapped tokens into Across and bridge them using remainder of input params.
+        IERC20(depositData.inputToken).safeApprove(address(spokePool), returnAmount);
```

Comment:
nit: we probably want to use safeIncreaseAllowance. safeApprove has been removed in the current version of [SafeERC20](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/master/contracts/token/ERC20/utils/SafeERC20.sol) because it has some footguns.